bigframes/session/__init__.py

```diff
@@ -249,27 +249,33 @@ def read_gbq(
 
         query_or_table = self._filters_to_query(query_or_table, columns, filters)
 
-        if _is_query(query_or_table):
-            return self._read_gbq_query(
-                query_or_table,
-                index_col=index_col,
-                columns=columns,
-                max_results=max_results,
-                api_name="read_gbq",
-                use_cache=use_cache,
-            )
-        else:
-            # TODO(swast): Query the snapshot table but mark it as a
-            # deterministic query so we can avoid serializing if we have a
-            # unique index.
-            return self._read_gbq_table(
-                query_or_table,
-                index_col=index_col,
-                columns=columns,
-                max_results=max_results,
-                api_name="read_gbq",
-                use_cache=use_cache,
-            )
+        try:
+
+            if _is_query(query_or_table):
+                return self._read_gbq_query(
+                    query_or_table,
+                    index_col=index_col,
+                    columns=columns,
+                    max_results=max_results,
+                    api_name="read_gbq",
+                    use_cache=use_cache,
+                )
+            else:
+                # TODO(swast): Query the snapshot table but mark it as a
+                # deterministic query so we can avoid serializing if we have a
+                # unique index.
+                return self._read_gbq_table(
+                    query_or_table,
+                    index_col=index_col,
+                    columns=columns,
+                    max_results=max_results,
+                    api_name="read_gbq",
+                    use_cache=use_cache,
+                )
+        except google.api_core.exceptions.Forbidden as ex:
+            if "Permission denied while getting Drive credentials." in ex.message:
```

Comment:
This could be a bit brittle. Let's only include the minimum to see if it's a Drive error.

```suggestion
            if "Drive credentials" in ex.message:
```

Reply:
Done, thanks! I also moved the exception handling deeper so that it serves mroe APIs. PTAL.